services/api-koa/src/resources/user/actions/remove.ts

```diff
@@ -0,0 +1,15 @@
+import userService from 'resources/user/user.service';
+import { AppKoaContext, AppRouter } from 'types';
+
+type ValidatedData = never;
+
+async function handler(ctx: AppKoaContext<ValidatedData>) {
+  await userService.removeSoft({ _id: ctx.request.params?.id });
```

Comment:
```suggestion
  await userService.removeSoft({ _id: ctx.request.params.id });
```
`id` always will be in params

Reply:
let's add validator that user exists

Reply:
>id always will be in params

According to types error it is possible the params can be `undefined`